Dockerfile

```diff
@@ -0,0 +1,34 @@
+FROM docker.io/openshift/origin-release:golang-1.10
+
+#install clang and gcc-c++ required for jsonnet to build
+RUN yum install -y clang \
+    gcc-c++
+
+#get required go packages for building k8s objects
+RUN go get github.com/jsonnet-bundler/jsonnet-bundler/cmd/jb \
+    github.com/brancz/gojsontoyaml
+
+#building jsonnet
+RUN git clone https://github.com/google/jsonnet && \
+    git --git-dir=jsonnet/.git checkout v0.10.0 && \
+    make -C jsonnet CC=clang CXX=clang++ && \
+    cp jsonnet/jsonnet /usr/bin
+
+#install kubectl package in container
+RUN curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl && \
+    chmod +x ./kubectl && \
+    mv ./kubectl /usr/local/bin
+
+#cloning the gluster-mixins project and making extras as working dir
+RUN git clone https://github.com/gluster/gluster-mixins /gluster/gluster-mixins;
```

Comment:
You want to build the container from the currently checked out artifacts, not from what exists in github, so this should be a `COPY` to grab the local version of the files instead of a git clone.

Reply:
Ack!

Reply:
`jsonnetfile.json` will still pull from the `latest github master`. 

If we are considering using local version for manifestation, we need to fix this dependency inside `extras` so that it pulls it's local parent for manifestation.

Reply:
Also, since we'll be working with local copy, i think it is better to add `promtool` and `make fmt` checks before building an image so that we don't end up in trouble later?

Reply:
@umangachapagain will be adding the test related parts to the docker image.